### ISSUE TITLE
Feature Request: Receive silent background updates of my own changes

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15537]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Problem:

Alice, Bob, and Cathy are chatting:
1. Alice sends message # 1 using desktop, which notifies Bob and Cathy
2. Bob responds with message # 2, which notifies Alice's mobile device
3. Alice taps Bob's notification and instantly sees message # 2 from Bob (because it is contained in the notification payload), but not her own message # 1 (because we do not send notifications for a person's own messages).

## Solution:
Every time we send a notification to Alice, walk backwards to also send every preceding message that was sent by Alice.  So for this:

1. Alice
2. Bob
3. Alice
4. Cathy
5. Bob

Then the notification to Alice for message # 2 would also include Alice's # 1.  Additionally, the notification to Alice for message# 4 would also include Alices# 3.  But message# 5 to Alice wouldn't include anything, because Cathy's message# 4 already generated its own notification. 

## Context/Examples/Screenshots/Notes:

**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1677480353745359

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013d1e50bf176ab73f</li>
        <li>Upwork Job ID: 1631093569840959488</li>
        <li>Last Price Increase: 2023-03-02</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @MitchExpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~013d1e50bf176ab73f

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor Plus for review of internal employee PR - @thesahindia (`Internal`)

**Comment by MitchExpensify** [comment]
Labelling Internal and Weekly based on [this SO](https://stackoverflowteams.com/c/expensify/questions/14418)

**Comment by Julesssss** [comment]
This is a bit problematic at the moment, as [we're severely limited](https://expensify.slack.com/archives/C01RJ1RCK8D/p1677782180232879?thread_ts=1676655235.172589&cid=C01RJ1RCK8D) by a Firebase 4kb limit to the notification payload data. 

I think this issue needs to be held on the outcome of that thread, which aims to strip out unnecessary data. Once that's done we can look intot using gzip or something similar in order to further reduce the package size, letting us bundle older messages into the notification payload. 

Another option that we can do in parallel is to run background tasks to pull data and persist into Onyx. But this is a bit tricky as the OS can and will restrict us from doing this as it kills the device battery/performance without the user being aware.

**Comment by Julesssss** [comment]
> Still held on that thread right @Julesssss ?

Yeah, more specifically it's held on [this issue.](https://github.com/Expensify/App/issues/14715) I'm currently working on it though so it shouldn't be much longer.

**Comment by MelvinBot** [comment]
@Julesssss @MitchExpensify @thesahindia this issue was created 2 weeks ago. Are we close to a solution? Let's make sure we're treating this as a top priority. Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by Julesssss** [comment]
Still held on the above issue. It's not a quick task but is in progress

**Comment by MelvinBot** [comment]
@Julesssss @MitchExpensify @thesahindia this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a>. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by MelvinBot** [comment]
@Julesssss, @MitchExpensify, @thesahindia Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MitchExpensify** [comment]
Progress is being made [here](https://github.com/Expensify/App/issues/14715) but we're still blocked on it

**Comment by Julesssss** [comment]
> Same as above

Yeah, I'm going to reduce the priority because there are least 2/3 issues that first need to be resolved.

**Comment by MitchExpensify** [comment]
How we lookin' here @Julesssss, dare I say "same as above" 😆

**Comment by Julesssss** [comment]
Yep, you guessed it. Manual Requests is a big time-hog right now, so this is still held by other notification issues.

**Comment by Julesssss** [comment]
Going to reduce to monthly temporarily

**Comment by MitchExpensify** [comment]
.. how temporary @Julesssss ? 😆

**Comment by Julesssss** [comment]
Good news, we've had a breakthrough on the onyxPayload issue mentioned here, so I can pick this up soon! However, it will likely be prioritised beneath those notifications [discussed here yesterday](https://expensify.slack.com/archives/C04E99DVBTP/p1685452984275419).

So probably a few more weeks, shouldn't be much longer than that.

**Comment by Julesssss** [comment]
The Airship API silent flag doesn't appear to be triggering the backgorund task. I started a thread here to discuss, and I have added logging [here](https://github.com/Expensify/App/pull/19613/files) in an attempt to debug the problem.

**Comment by MitchExpensify** [comment]
How's the debugging going @Julesssss ?

**Comment by Julesssss** [comment]
I'm back on the task this week. While working on [this issue](https://github.com/Expensify/App/pull/19613/files) I have been introducing better logging so that I can better debug the background service issue.

**Comment by Julesssss** [comment]
Made progress today [here](https://github.com/Expensify/Expensify/issues/260137)

**Comment by Julesssss** [comment]
Picking this one up soon, now that the [notification threading PR](https://github.com/Expensify/Expensify/issues/260137) is in review

**Comment by MitchExpensify** [comment]
How are we looking on this one @Julesssss ?

**Comment by Julesssss** [comment]
Due to a [change in priorities](https://expensify.slack.com/archives/C04E99DVBTP/p1689098539376669), I am once again switching to the Gzipping payloads issue. This issue will likely follow that one.

**Comment by Julesssss** [comment]
No update, remaining at a low priority

**Comment by Julesssss** [comment]
Yes sir

**Comment by Julesssss** [comment]
Reducing priority again. We closed out the reliable updates project and listed this as follow-up polish/improvement.

**Comment by Julesssss** [comment]
We're not ready to look into this yet.

**Comment by Julesssss** [comment]
Kinda tempted to close this. It's likely a 2024 project.

**Comment by Julesssss** [comment]
Given our focus change, I'm thinking we might have to close this issue for a while. What do you think @arosiclair ?

**Comment by arosiclair** [comment]
Yeah that sounds fair to me. We can leave it in the list and take another look later if we still think we could use this
